circus-train-aws/src/test/java/com/hotels/bdp/circustrain/aws/S3DataManipulatorTest.java

```diff
@@ -0,0 +1,105 @@
+import static org.junit.Assert.*;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.hadoop.fs.s3a.BasicAWSCredentialsProvider;
+import org.gaul.s3proxy.junit.S3ProxyRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+
+@RunWith(MockitoJUnitRunner.class)
+public class S3DataManipulatorTest {
+
+  private static final String AWS_ACCESS_KEY = "access";
+  private static final String AWS_SECRET_KEY = "secret";
+  private static final String BUCKET = "bucket";
+  private static final String BUCKET_PATH = "s3://" + BUCKET;
+  private static final String FOLDER = "folder";
+  private static final String EMPTY_BUCKET = "empty-bucket";
+
+  public @Rule TemporaryFolder temp = new TemporaryFolder();
+  public @Rule S3ProxyRule s3Proxy = S3ProxyRule.builder().withCredentials(AWS_ACCESS_KEY, AWS_SECRET_KEY).build();
+
+  private S3DataManipulator s3DataManipulator;
+  private AmazonS3 s3Client;
+
+  @Before
+  public void setUp() throws IOException {
+    s3Client = newClient();
+    s3DataManipulator = new S3DataManipulator(s3Client);
+    s3Client.createBucket(BUCKET);
+    File inputData = temp.newFile("data");
+    s3Client.putObject(BUCKET, FOLDER, inputData);
+  }
+
+  private AmazonS3 newClient() {
+    AwsClientBuilder.EndpointConfiguration endpointConfiguration = new AwsClientBuilder.EndpointConfiguration(s3Proxy.getUri().toString(),
+        Regions.DEFAULT_REGION.getName());
+    AmazonS3 newClient = AmazonS3ClientBuilder
+        .standard()
+        .withCredentials(new BasicAWSCredentialsProvider(AWS_ACCESS_KEY, AWS_SECRET_KEY))
+        .withEndpointConfiguration(endpointConfiguration)
+        .build();
+    return newClient;
+  }
+
+  @Test
+  public void deleteFolderSucceeds() {
+    boolean result = s3DataManipulator.delete(BUCKET_PATH + "/" + FOLDER);
+    assertThat(result, is(true));
+  }
+
+  @Test
+  public void deleteNonexistentFolderFails() {
+    boolean result = s3DataManipulator.delete(BUCKET_PATH + "/nonexistent-folder");
+    assertThat(result, is(false));
+  }
+
+  @Test
+  public void deleteBucketSucceeds() {
+    boolean result = s3DataManipulator.delete(BUCKET_PATH);
+    assertThat(result, is(true));
+  }
+
+  @Test
+  public void deleteEmptyBucketFails() {
+    s3Client.createBucket(EMPTY_BUCKET);
+    boolean result = s3DataManipulator.delete("s3://" + EMPTY_BUCKET);
+    assertThat(result, is(false));
+  }
+
+  @Test(expected = AmazonS3Exception.class)
+  public void deleteNonExistentBucketThrowsException() {
+    boolean result = s3DataManipulator.delete("s3://" + "nonexistent-bucket");
```

Comment:
If this throws an exception you don't need to store the result or assert on it as that code will never execute.

Reply:
true, copy pasted and forgot to update 😄